impl/src/main/java/org/pentaho/platform/genericfile/providers/repository/RepositoryFileProvider.java

```diff
@@ -492,6 +492,15 @@ public List<IGenericFile> getDeletedFiles() {
       .collect( Collectors.toList() );
   }
 
+  @Override
+  public void permanentDeleteFiles( @NonNull List<String> fileIds ) throws OperationFailedException {
+    try {
+      fileService.doDeleteFilesPermanent( String.join( ",", fileIds ) );
```

Comment:
One way to avoid the situation of partial failure would be to test delete permissions to all the provided files before trying to delete them. Maybe this could be done directly by `DefaultGenericFileService`, before even calling the providers' delete method, by calling the hasAccess method.

Reply:
We ended up deciding to let each delete operation test this, as these already do, and have the service have a fully "do as many as possible" approach.